packages/core-products/src/module/configureProductsModule.ts

```diff
@@ -361,10 +362,29 @@ export const configureProductsModule = async ({
 
       return productId;
     },
-
+    firstActiveProductLink: async (productId) => {
+      const activeLink = await Products.findOne(
+        {
+          $or: [{ 'proxy.assignments.productId': productId }, { 'bundleItems.productId': productId }],
+          status: { $ne: 'DELETED' },
+        },
+        { projection: { _id: 1, proxy: 1, bundleItems: 1 } },
+      );
+
+      return {
+        _id: activeLink?._id,
+        variations:
+          activeLink?.proxy?.assignments?.filter(
+            ({ productId: variantProductId }) => variantProductId === productId,
+          ) || [],
+        bundles:
+          activeLink?.bundleItems?.filter(
+            ({ productId: bundledProductId }) => bundledProductId === productId,
+          ) || [],
+      };
```

Comment:
Strongly suggest to make 2 functions, this is mixing stuff that has no relation to each other